### PR TITLE
".fap" extention in file browser and archive tab

### PR DESCRIPTION
**Pull request by drunkbatya**
# What's new

- Add file manager associated extension ".fap"
- Add archive tab "Apps" ("Applications" doesn't fit in title frame)
- Loading icons and package name from ".fap"

# Verification 

- [ Describe how to verify changes ]

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Review by hedger — DISMISSED** [review]

File: applications/main/archive/helpers/archive_browser.c [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include <core/common_defines.h>
 #include <core/log.h>
 #include "gui/modules/file_browser_worker.h"
+#include "../../fap_loader/fap_loader_app.h"
```
Comment (DrZlo13) [inline applications/main/archive/helpers/archive_browser.c]: `#include <fap_loader/fap_loader_app.h>`

File: applications/main/archive/helpers/archive_browser.c [hunk 2]
```diff
@@ -351,15 +352,34 @@ void archive_add_app_item(ArchiveBrowserView* browser, const char* name) {
     ArchiveFile_t_clear(&item);
 }
 
+static bool archive_get_fap_meta(string_t file_path, string_t fap_name, uint8_t** icon_ptr) {
+    FapLoader* loader = malloc(sizeof(FapLoader));
```
Comment (DrZlo13) [inline applications/main/archive/helpers/archive_browser.c]: App internals should not be available outside. Implement `fap_loader_alloc();` and `fap_loader_free();`.

File: applications/main/archive/helpers/archive_browser.c (hunk 1 above)
Comment (DrZlo13) [inline applications/main/archive/helpers/archive_browser.c]: `#include <fap_loader/fap_loader_app.h>`

File: applications/main/archive/helpers/archive_browser.c [hunk 3]
```diff
@@ -351,15 +352,34 @@ void archive_add_app_item(ArchiveBrowserView* browser, const char* name) {
...
+    free(loader);
+    return success;
+}
+
 void archive_add_file_item(ArchiveBrowserView* browser, bool is_folder, const char* name) {
     furi_assert(browser);
     furi_assert(name);
 
     ArchiveFile_t item;
+    item.custom_icon_data = NULL;
```
Comment (DrZlo13) [inline applications/main/archive/helpers/archive_browser.c]: ArchiveFile_t has an initializer, ArchiveFile_t_init.

File: applications/main/archive/helpers/archive_browser.c [hunk 4]
```diff
@@ -351,15 +352,34 @@ void archive_add_app_item(ArchiveBrowserView* browser, const char* name) {
...
 
     ArchiveFile_t_init(&item);
     string_init_set_str(item.path, name);
+    string_init(item.display_name);
```
Comment (DrZlo13) [inline applications/main/archive/helpers/archive_browser.c]: ArchiveFile_t has an initializer, ArchiveFile_t_init.

File: applications/main/archive/helpers/archive_browser.c [hunk 5]
```diff
@@ -351,15 +352,34 @@ void archive_add_app_item(ArchiveBrowserView* browser, const char* name) {
...
+
 void archive_add_file_item(ArchiveBrowserView* browser, bool is_folder, const char* name) {
     furi_assert(browser);
     furi_assert(name);
 
     ArchiveFile_t item;
+    item.custom_icon_data = NULL;
 
     ArchiveFile_t_init(&item);
     string_init_set_str(item.path, name);
```
Comment (DrZlo13) [inline applications/main/archive/helpers/archive_browser.c]: memleak, string already initialized inside ArchiveFile_t_init. Needs to be checked in other places.

File: applications/main/archive/helpers/archive_files.h [hunk 6]
```diff
@@ -4,6 +4,8 @@
 #include <m-string.h>
 #include <storage/storage.h>
 
+#define CUSTOM_ICON_MAX_SIZE 32
```
Comment (DrZlo13) [inline applications/main/archive/helpers/archive_files.h]: FAP_MANIFEST_MAX_ICON_SIZE

File: applications/main/archive/helpers/archive_files.h [hunk 7]
```diff
@@ -21,33 +24,56 @@ typedef enum {
 typedef struct {
     string_t path;
     ArchiveFileTypeEnum type;
+    uint8_t* custom_icon_data;
+    string_t display_name;
```
Comment (DrZlo13) [inline applications/main/archive/helpers/archive_files.h]: custom_name

File: applications/main/archive/views/archive_browser_view.c [hunk 8]
```diff
@@ -124,12 +127,23 @@ static void draw_list(Canvas* canvas, ArchiveBrowserViewModel* model) {
...
                 model->files, CLAMP(idx - model->array_offset, (int32_t)(array_size - 1), 0));
-            path_extract_filename(file->path, str_buf, archive_is_known_app(file->type));
             file_type = file->type;
+            if(file_type == ArchiveFileTypeApplication) {
+                if(file->custom_icon_data) {
+                    custom_icon_data = file->custom_icon_data;
+                    string_set(str_buf, file->display_name);
+                } else {
+                    file_type = ArchiveFileTypeUnknown;
+                    path_extract_filename(file->path, str_buf, archive_is_known_app(file->type));
```
Comment (DrZlo13) [inline applications/main/archive/views/archive_browser_view.c:141]: move path_extract_filename outside `if`

File: applications/main/fap_loader/fap_loader_app.h [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+typedef struct {
+    FlipperApplication* app;
+    Storage* storage;
+    DialogsApp* dialogs;
+    Gui* gui;
+    string_t fap_path;
+
+    ViewDispatcher* view_dispatcher;
+    Loading* loading;
+} FapLoader;
```
Comment (DrZlo13) [inline applications/main/fap_loader/fap_loader_app.h]: App internals should not be available outside.

**Review by DrZlo13 — CHANGES_REQUESTED** [review]

File: applications/main/archive/views/archive_browser_view.c (hunk 8 above)
Comment (drunkbatya) [reply applications/main/archive/views/archive_browser_view.c:141]: I think path_extract_filename call isn't needed if we can load icon from .fap

File: applications/main/archive/helpers/archive_files.h [hunk 10]
```diff
@@ -21,33 +24,56 @@ typedef enum {
 typedef struct {
     FuriString* path;
     ArchiveFileTypeEnum type;
+    uint8_t* custom_icon_data;
+    FuriString* custom_name;
     bool fav;
     bool is_app;
```
Comment (DrZlo13) [inline applications/main/archive/helpers/archive_files.h:30]: Variable `is_app` is not used

**Review by DrZlo13 — CHANGES_REQUESTED** [review]

**Review by DrZlo13 — APPROVED** [review]

**Review by skotopes — APPROVED** [review]

**Pull request merged**